drivers/counter/counter_ambiq_timer.c

```diff
@@ -16,59 +16,250 @@
+	default:
+		return AM_HAL_CTIMER_FN_ONCE;
+	}
+}
+
+static uint32_t get_ambiq_interrupt_bit(uint32_t timer)
+{
+	switch (timer) {
+	case 0:
+		return AM_HAL_CTIMER_INT_TIMERA0C0;
+	case 1:
+		return AM_HAL_CTIMER_INT_TIMERA1C0;
+	case 2:
+		return AM_HAL_CTIMER_INT_TIMERA2C0;
+	case 3:
+		return AM_HAL_CTIMER_INT_TIMERA3C0;
+	case 4:
+		return AM_HAL_CTIMER_INT_TIMERA4C0;
+	case 5:
+		return AM_HAL_CTIMER_INT_TIMERA5C0;
+	case 6:
+		return AM_HAL_CTIMER_INT_TIMERA6C0;
+	case 7:
+		return AM_HAL_CTIMER_INT_TIMERA7C0;
+	default:
+		return AM_HAL_CTIMER_INT_TIMERA0C0;
+	}
+}
+
+static uint32_t get_ambiq_clk(uint32_t clk)
+{
+	switch (clk) {
+	case 0:
+		return AM_HAL_CTIMER_CLK_PIN;
+	case 1:
+		return AM_HAL_CTIMER_HFRC_12MHZ;
+	case 2:
+		return AM_HAL_CTIMER_HFRC_3MHZ;
+	case 3:
+		return AM_HAL_CTIMER_HFRC_187_5KHZ;
+	case 4:
+		return AM_HAL_CTIMER_HFRC_47KHZ;
+	case 5:
+		return AM_HAL_CTIMER_HFRC_12KHZ;
+	case 6:
+		return AM_HAL_CTIMER_XT_32_768KHZ;
+	case 7:
+		return AM_HAL_CTIMER_XT_16_384KHZ;
+	case 8:
+		return AM_HAL_CTIMER_XT_2_048KHZ;
+	case 9:
+		return AM_HAL_CTIMER_XT_256HZ;
+	case 10:
+		return AM_HAL_CTIMER_LFRC_512HZ;
+	case 11:
+		return AM_HAL_CTIMER_LFRC_32HZ;
+	case 12:
+		return AM_HAL_CTIMER_LFRC_1HZ;
+	case 13:
+		return AM_HAL_CTIMER_LFRC_1_16HZ;
+	case 14:
+		return AM_HAL_CTIMER_RTC_100HZ;
+	default:
+		return AM_HAL_CTIMER_CLK_PIN;
+	}
+}
 
+static uint32_t get_ambiq_clk_ctrl(uint32_t clk)
+{
+	switch (clk) {
+	case AM_HAL_CTIMER_CLK_PIN:
+		return -ENOTSUP;
+	case AM_HAL_CTIMER_HFRC_12MHZ:
+	case AM_HAL_CTIMER_HFRC_3MHZ:
+	case AM_HAL_CTIMER_HFRC_187_5KHZ:
+	case AM_HAL_CTIMER_HFRC_47KHZ:
+	case AM_HAL_CTIMER_HFRC_12KHZ:
+		return AM_HAL_CLKGEN_CONTROL_SYSCLK_MAX;
+	case AM_HAL_CTIMER_XT_32_768KHZ:
+	case AM_HAL_CTIMER_XT_16_384KHZ:
+	case AM_HAL_CTIMER_XT_2_048KHZ:
+	case AM_HAL_CTIMER_XT_256HZ:
+		return AM_HAL_CLKGEN_CONTROL_XTAL_START;
+	case AM_HAL_CTIMER_LFRC_512HZ:
+	case AM_HAL_CTIMER_LFRC_32HZ:
+	case AM_HAL_CTIMER_LFRC_1HZ:
+	case AM_HAL_CTIMER_LFRC_1_16HZ:
+		return AM_HAL_CLKGEN_CONTROL_LFRC_START;
+	case AM_HAL_CTIMER_RTC_100HZ:
+		return -ENOTSUP;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static uint32_t get_freq(uint32_t clk)
+{
+	switch (clk) {
+	case AM_HAL_CTIMER_CLK_PIN:
+		return 0;
+	case AM_HAL_CTIMER_HFRC_12MHZ:
+		return 12000000;
+	case AM_HAL_CTIMER_HFRC_3MHZ:
+		return 3000000;
+	case AM_HAL_CTIMER_HFRC_187_5KHZ:
+		return 1875000;
+	case AM_HAL_CTIMER_HFRC_47KHZ:
+		return 47000;
+	case AM_HAL_CTIMER_HFRC_12KHZ:
+		return 12000;
+	case AM_HAL_CTIMER_XT_32_768KHZ:
+		return 32768;
+	case AM_HAL_CTIMER_XT_16_384KHZ:
+		return 16384;
+	case AM_HAL_CTIMER_XT_2_048KHZ:
+		return 2048;
+	case AM_HAL_CTIMER_XT_256HZ:
+		return 256;
+	case AM_HAL_CTIMER_LFRC_512HZ:
+		return 512;
+	case AM_HAL_CTIMER_LFRC_32HZ:
+		return 32;
+	case AM_HAL_CTIMER_LFRC_1HZ:
+		return 1;
+	case AM_HAL_CTIMER_LFRC_1_16HZ:
+		return 1;
+	case AM_HAL_CTIMER_RTC_100HZ:
+		return 100;
+	default:
+		return 32768;
+	}
+};
```

Comment:
Could these be ifdef'd based on CONFIG_SOC_APOLLO3P_BLUE as well or moved somewhere else it makes sense? I'd expect the build to complain if we weren't building with that config enabled.

Reply:
Done